packages/snap/src/index.tsx

```diff
@@ -91,3 +102,50 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
       throw new Error('Method not found.');
   }
 };
+
+/**
+ * Handle incoming home page requests from the MetaMask clients.
+ *
+ * @returns A static panel rendered with custom UI.
+ * @see https://docs.metamask.io/snaps/reference/exports/#onhomepage
+ */
+export const onHomePage: OnHomePageHandler = async () => {
+  const network = await getNetwork();
+  const account = await getHDAccount();
+  const balance = await getBalance({ address: account.address || '' });
+
+  return {
+    content: (
+      <HomePage
+        network={network.network}
+        address={account.address || ''}
+        balance={toMAS(balance.candidateBalance).toString()}
+      />
+    ),
+  };
+};
+
+/**
+ * Handle incoming user events coming from the Snap interface.
+ *
+ * @param params - The event parameters.
+ * @param params.id - The Snap interface ID where the event was fired.
+ * @param params.event - The event object containing the event type, name and
+ * value.
+ * @see https://docs.metamask.io/snaps/reference/exports/#onuserinput
+ */
+export const onUserInput: OnUserInputHandler = async ({ event, id }) => {
+  assert(event.type === UserInputEventType.ButtonClickEvent);
+  assert(event.name === 'show-keys-validation' || 'show-keys');
```

Comment:
this assert is wrong
`'show-keys-validation' || 'show-keys'`always evaluate to "true"
what you want:
assert(event.name === 'show-keys-validation' || event.name === 'show-keys');